zproject/default_settings.py

```diff
@@ -199,29 +199,47 @@
 EMBEDDED_BOTS_ENABLED = False
 
 DEFAULT_RATE_LIMITING_RULES = {
+    # Applies to API requests in authenticated contexts,
+    # each user gets their own bucket.
     "api_by_user": [
         (60, 200),  # 200 requests max every minute
     ],
+    # Applies to API requests in unauthenticated contexts
+    # (web public views) - ONLY if the request is not authenticated.
     "api_by_ip": [
         (60, 100),
     ],
+    # Applies to requests to our push bouncer API, by an authenticated
+    # remote server.
     "api_by_remote_server": [
         (60, 1000),
     ],
+    # Limits how many authentication attempts with login+password can be made
+    # to a single username. This applies to the authentication backends where login+password
+    # gets submitted to the Zulip server - the ldap backend or regular email+password backend.
     "authenticate_by_username": [
         (1800, 5),  # 5 failed login attempts within 30 minutes
     ],
+    # Limits how many requests to change the account email address
+    # can be made per user.
     "email_change_by_user": [
         (3600, 2),  # 2 per hour
         (86400, 5),  # 5 per day
     ],
+    # Limits how many requests to send the password reset email
+    # can be made for a single email address.
     "password_reset_form_by_email": [
         (3600, 2),  # 2 reset emails per hour
         (86400, 5),  # 5 per day
     ],
+    # This limit applies to all requests which directly trigger the sending of an email,
+    # restricting the number per IP address.
     "sends_email_by_ip": [
         (86400, 5),
     ],
+    # Limits access to uploaded files, in web-public contexts, done by
+    # unauthenticated users. Each file gets its own bucket, and every access
+    # to the file by an unauthenticated user counts towards the limit.
```

Comment:
Do we want to clarify that this is to prevent unauthenticated web-public files from being used as file-sharing?

Reply:
Good idea!